core/store/src/main/java/mil/nga/giat/geowave/core/store/operations/config/AddStoreCommand.java

```diff
@@ -114,11 +111,46 @@ public void execute(
 					"That store already exists: " + getPluginName());
 		}
 
+		if (pluginOptions.getFactoryOptions() != null) {
+			pluginOptions.getFactoryOptions().validatePluginOptions(
+					existingProps);
+		}
+
 		// Save the store options.
 		pluginOptions.save(
 				existingProps,
 				getNamespace());
 
+		if (pluginOptions.getFactoryOptions() != null) {
```

Comment:
add this logic to write properties file method (and decrypt immediately on read).  those should be the only 2 places where we need to worry about encrypt/decrypt (as far as I know, I could stand corrected).